src/univariates.jl

```diff
@@ -37,6 +37,10 @@ end
 @compat support{D<:ContinuousUnivariateDistribution}(d::Union{D,Type{D}}) = RealInterval(minimum(d), maximum(d))
 @compat support{D<:DiscreteUnivariateDistribution}(d::Union{D,Type{D}}) = round(Int, minimum(d)):round(Int, maximum(d))
 
+# Type used for dispatch on finite support
+# T = true or false
+immutable FiniteSupport{T} end
```

Comment:
I think this might be overdoing it. Each call to `pdf` is likely to be quite a bit more expensive than a branch in `cdf`. This package is quite heavy on compilation already so we should consider avoiding extra compilation when it doesn't gives a speedup or greatly simplifies the code.

It would also be good to wait at least 24 hours for comments. Especially on a the Weekend. Except for these two comments, I think the changes are fine.